### PR TITLE
feat(backend): accept OAuth bearer token

### PR DESCRIPTION
**Pull request by saschanaz**
## What

`Authorization: Bearer` headerで認証できるように

## Why

#8262 に必要

WWW-Authenticateは https://datatracker.ietf.org/doc/html/rfc6750.html#section-3 と https://www.rfc-editor.org/rfc/rfc9110#section-15.5.2 での要求事項

## Checklist
- [ ] Read the [contribution guide](https://github.com/misskey-dev/misskey/blob/develop/CONTRIBUTING.md)
- [ ] Test working in a local environment
- [ ] (If needed) Add story of storybook
- [ ] (If needed) Update CHANGELOG.md
- [ ] (If possible) Add tests

### CODE REVIEW

File: packages/backend/src/server/api/ApiCallService.ts [hunk 1]
```diff
@@ -58,23 +58,28 @@ export class ApiCallService implements OnApplicationShutdown {
...
-	) {
+	): void {
 		const body = request.method === 'GET'
 			? request.query
 			: request.body;
 
-		const token = body?.['i'];
+		// https://datatracker.ietf.org/doc/html/rfc6750.html#section-2.1 (case sensitive)
+		const token = request.headers.authorization?.startsWith('Bearer ') ?
+			request.headers.authorization.slice(7) : body?.['i'];
```
Comment (acid-chicken) [inline packages/backend/src/server/api/ApiCallService.ts]:
```suggestion
		const token = request.headers.authorization?.startsWith('Bearer ')
			? request.headers.authorization.slice(7)
			: body?.['i'];
```

File: packages/backend/src/server/api/StreamingApiServerService.ts [hunk 2]
```diff
@@ -58,11 +58,20 @@ export class StreamingApiServerService {
 			let user: LocalUser | null = null;
 			let app: AccessToken | null = null;
 
+			// https://datatracker.ietf.org/doc/html/rfc6750.html#section-2.1
+			// Note that the standard WHATWG WebSocket API does not support setting any headers,
+			// but non-browser apps may still be able to set it.
+			const token = request.headers.authorization?.startsWith('Bearer ') ?
+				request.headers.authorization.slice(7) : q.get('i');
```
Comment (acid-chicken) [inline packages/backend/src/server/api/StreamingApiServerService.ts]:
```suggestion
			const token = request.headers.authorization?.startsWith('Bearer ')
				? request.headers.authorization.slice(7)
				: q.get('i');
```

File: packages/backend/src/server/api/ApiCallService.ts [hunk 3]
```diff
@@ -117,7 +124,9 @@ export class ApiCallService implements OnApplicationShutdown {
 			fields[k] = typeof v === 'object' && 'value' in v ? v.value : undefined;
 		}
 
-		const token = fields['i'];
+		// https://datatracker.ietf.org/doc/html/rfc6750.html#section-2.1 (case sensitive)
+		const token = request.headers.authorization?.startsWith('Bearer ') ?
+			request.headers.authorization.slice(7) : fields['i'];
```
Comment (acid-chicken) [inline packages/backend/src/server/api/ApiCallService.ts]:
```suggestion
		const token = request.headers.authorization?.startsWith('Bearer ')
			? request.headers.authorization.slice(7)
			: fields['i'];
```

**Review by acid-chicken — APPROVED** [review]
コード自体は良さそう
OAuth じゃないトークンを Bearer で使えてしまうのは良い？

**Comment by saschanaz** [comment]
> OAuth じゃないトークンを Bearer で使えてしまうのは良い？

https://datatracker.ietf.org/doc/html/rfc6750.html#section-1

> While designed for use with access tokens resulting from OAuth 2.0 authorization [[RFC6749](https://datatracker.ietf.org/doc/html/rfc6749)] flows to access OAuth protected resources, this specification actually defines a general HTTP authorization method that can be used with bearer tokens from any source to access any resources protected by those bearer tokens.

問題なさそうです

**Comment by syuilo** [comment]
👍🏻

**Pull request merged**